department-of-veterans-affairs/caseflow
# Kamalam7/appeals 13366

Resolves APPEALS-13366

Description
Created Proceed to final notification letter task modal. From Send Initial Notification Letter task action, select Proceed to final letter from the dropdown. Fill in the instructions in the text area and click on Proceed to final letter. Case appears in the Currently active task as Send Final Notification Letter task and in the Case timeline Send Initial Notification Letter Task is marked as Completed. The case appears in the COB Organization view under Unassigned Tab with Task as Send Final Notification Letter Task.

Acceptance Criteria


- [x] The following occurs when the “Proceed to Final Notification Letter” task action is selected for the Send Initial Notification Letter task
- [x] Modal appears asking user to confirm that they want to proceed to final notification letter. User is required to provide a reason why they want to proceed to final notification letter, Field type: free text with a 350-character limit 
- [x] The following occurs when user executes the “Proceed to Final Notification Letter” task action for the Send Initial Notification Letter task
          The “Send Initial Notification Letter” task is marked as Complete
          Task moves from Currently Active Tasks section to Case Timeline and contains the 
          following information
                  Date of occurrence
                  Assigned to: <organization>
                  Assigned by: <user who completed action to assign task to COB>
                  Task Instructions: <user input in the Proceed to final notification letter modal>

- [x] The “Send Final Notification Letter” task is created and active
                Task appears in the Currently Active Tasks section
                Assigned to the COB organization
                Case appears on the Unassigned tab of the COB Queue


Testing Plan
1. Go to Caseflow Login and Sign as BVADWISE
2. Navigate to Mail Intake
3. From the dropdown select VA FORM 10182 for processing and click on continue to search
![image](https://user-images.githubusercontent.com/110078646/217350491-72eb791c-9b0d-42c3-87db-913e290e9ab2.png)


4. Enter a veteran file number. I used 578270010. Click the search icon to continue
![image](https://user-images.githubusercontent.com/110078646/217351593-d69f17b0-3082-4611-83c9-c52139ccf656.png)

5. Fill out the form and select direct review for the review option type. Click the button at the bottom to continue
<img width="871" alt="image" src="https://user-images.githubusercontent.com/110078646/217352023-2f5be85a-35ee-4142-9dff-a298398c52e1.png">

6. On the add issues page, click the add issues button. Select a benefit type as Compensation and Issue Category as "Contested claims - Attorney Fees. Click on Add this issue and establish appeal to complete the intake.
![image](https://user-images.githubusercontent.com/110078646/217353825-60ffc401-dd31-4cdf-85d2-007c8309fdac.png)

7. The intake should be completed. Copy the veteran id and click search cases.
<img width="879" alt="image" src="https://user-images.githubusercontent.com/110078646/217354063-7ba9fe19-0738-4f4a-b5c5-8513ccce649d.png">

8. Enter the veteran file number into the search bar and select search. The issue you are looking for will have a purple CC icon next to it. Click the docket number link to get to the veteran's case details page.
<img width="884" alt="image" src="https://user-images.githubusercontent.com/110078646/217354603-69f590bb-74b4-4b14-831f-d6591921af29.png">

9. In the currently active tasks, the Send Initial Notification Letter should show and be assigned to COB. No actions should display.
<img width="881" alt="image" src="https://user-images.githubusercontent.com/110078646/217354889-9ade7a49-79a3-462f-8c22-b6001a6a5806.png">

10. To see the action list, you will need to sign in as a COB member. Click the user name in the right corner and switch users from the dropdown. On the user login page, sign in as a COB user.

<img width="863" alt="image" src="https://user-images.githubusercontent.com/110078646/217355443-4e90543d-27cf-4a39-82c3-dd6bdd1e7270.png">

<img width="887" alt="image" src="https://user-images.githubusercontent.com/110078646/217355592-c6d191f2-e6f1-4524-ab17-abfd60329c67.png">

11. In queue, as a COB Organization you will find the COB View and under Unassigned tab, the contested claim appeal with Send Initial Notification Letter Task is shown under the Case details. 
<img width="871" alt="image" src="https://user-images.githubusercontent.com/110078646/217355803-88a7cea7-dd57-4bc3-a336-4842282b8715.png">

12. Click on the user under the Case details from the COB view, in the Currently active task you will find the Task as Send Initial Notification Letter, assigned to COB Organization and assigned by BVA Intake team.

<img width="863" alt="image" src="https://user-images.githubusercontent.com/110078646/217356016-d8021ce4-d5bb-4ce5-8acc-4b0040af4af2.png">


13. In the Currently active task, actions you will find the dropdown where you can select an action as Mark task complete, Proceed to final and Cancel task. 

<img width="863" alt="image" src="https://user-images.githubusercontent.com/110078646/217356111-4c028a5c-9345-48e5-83aa-25365242814e.png">

14. From the Actions, when you select Proceed to final, the Proceed to final notification letter modal appears. Enter the instructions in the text area and click on Proceed to final.

<img width="879" alt="image" src="https://user-images.githubusercontent.com/110078646/217356261-a022eebf-5af0-41e9-bdb0-49674c6e84d2.png">

15. The case is updated under the unassigned tab with the Task as Send Final Notification Letter.
<img width="866" alt="image" src="https://user-images.githubusercontent.com/110078646/217356558-9db96626-4671-4aa0-8dc3-5dac118532aa.png">

16. In the Currently active task you can see Task as Send Final Notification Letter. In the Case timeline you can see the Send Initial Notification Letter task completed.
<img width="878" alt="image" src="https://user-images.githubusercontent.com/110078646/217356694-04495d98-e65c-4de6-bf30-86162a6914b8.png">

<img width="878" alt="image" src="https://user-images.githubusercontent.com/110078646/217356774-f9a595ec-f05d-48b7-8203-9db467979ae1.png">
